apps/email/js/cards/message_list.js

```diff
@@ -443,10 +443,10 @@ return [
         this.messagesContainer.classList.add('show-edit');
 
         this.selectedMessages = [];
-        var cbs = this.messagesContainer
-                  .querySelectorAll('input[type=checkbox]');
-        for (i = 0; i < cbs.length; i++) {
-          cbs[i].checked = false;
+        var msgNodes = this.messagesContainer.querySelectorAll(
+          '.msg-header-item');
+        for (i = 0; i < msgNodes.length; i++) {
+          this.setMessageChecked(msgNodes[i], false);
```

Comment:
I think we should be doing this reset of the message checked to be false for all nodes, regardless of `if (editMode) {`. The else case, when editMode is false, is doing some work with 'msg-header-item-selected' that I believe is the same attempt at a reset, but based on very old code. So we can just remove that ''msg-header-item-selected' work and do this setMessageChecked false work outside the if/else.

Otherwise, going back to non-edit mode by clicking x will leave some msgNodes with aria-selected still as true.